src/components/service-metrics/service-metrics.ts

```diff
@@ -51,142 +73,165 @@ export async function resolveServiceMetrics(ctx: IContext, params: IParameters):
-        cf.userServices(params.spaceGUID),
-        cf.space(params.spaceGUID),
-        cf.organization(params.organizationGUID),
-    ]);
-
-    const isUserProvidedService = userProvidedServices.some(s => s.metadata.guid === params.serviceGUID);
-
-    const service = isUserProvidedService ?
-      await cf.userServiceInstance(params.serviceGUID) :
-      await cf.serviceInstance(params.serviceGUID);
-
-    const serviceLabel = isUserProvidedService ? 'User Provided Service'
-        : (await cf.service(service.entity.service_guid)).entity.label;
-
-    const breadcrumbs: ReadonlyArray<IBreadcrumb> = fromOrg(ctx, organization, [
-        {
-            text: space.entity.name,
-            href: ctx.linkTo('admin.organizations.spaces.services.list', {
-            organizationGUID: organization.metadata.guid,
-            spaceGUID: space.metadata.guid,
-            }),
-        },
-        { text: service.entity.name },
-    ]);
-
-    const cloudWatchMetricDataClient = new CloudwatchMetricDataClient(
-      new cw.CloudWatchClient({
-        region: ctx.app.awsRegion,
-        endpoint: ctx.app.awsCloudwatchEndpoint,
-      }),
-      new cw.CloudWatchClient({
-        region: 'us-east-1',
-        endpoint: ctx.app.awsCloudwatchEndpoint,
-      }),
-      new rg.ResourceGroupsTaggingAPIClient({
-        region: 'us-east-1',
-        endpoint: ctx.app.awsResourceTaggingAPIEndpoint,
+  const cf = new CloudFoundryClient({
+    accessToken: ctx.token.accessToken,
+    apiEndpoint: ctx.app.cloudFoundryAPI,
+    logger: ctx.app.logger,
+  });
+
+  const isAdmin = ctx.token.hasAnyScope(
+    CLOUD_CONTROLLER_ADMIN,
+    CLOUD_CONTROLLER_READ_ONLY_ADMIN,
+    CLOUD_CONTROLLER_GLOBAL_AUDITOR,
+  );
+
+  if (!params.rangeStart || !params.rangeStop) {
+    return {
+      status: 302,
+      redirect: ctx.linkTo(
+        'admin.organizations.spaces.services.metrics.view', {
+          organizationGUID: params.organizationGUID,
+          spaceGUID: params.spaceGUID,
+          serviceGUID: params.serviceGUID,
+          rangeStart: moment().subtract(24, 'hours').format('YYYY-MM-DD[T]HH:mm'),
+          rangeStop: moment().format('YYYY-MM-DD[T]HH:mm'),
+        }),
+    };
+  }
+
+  const {rangeStart, rangeStop, period} = parseRange(params.rangeStart, params.rangeStop);
+
+  const [isManager, isBillingManager, userProvidedServices, space, organization] = await Promise.all([
+    cf.hasOrganizationRole(params.organizationGUID, ctx.token.userID, 'org_manager'),
+    cf.hasOrganizationRole(params.organizationGUID, ctx.token.userID, 'billing_manager'),
+    cf.userServices(params.spaceGUID),
+    cf.space(params.spaceGUID),
+    cf.organization(params.organizationGUID),
+  ]);
+
+  const isUserProvidedService = userProvidedServices.some(s => s.metadata.guid === params.serviceGUID);
+
+  const service = isUserProvidedService ?
+  await cf.userServiceInstance(params.serviceGUID) :
+  await cf.serviceInstance(params.serviceGUID);
```

Comment:
Can we indent the `await _____` statements within this ternary block so that it's easier to read?